### PR TITLE
Docs and Tests for "gpus" Trainer Argument

### PR DESCRIPTION
**Pull request by awaelchli**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/williamFalcon/pytorch-lightning/blob/master/.github/CONTRIBUTING.md)?
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  

## What does this PR do?
Updates docs for the `gpus` Trainer argument and adds tests for parsing it as discussed in  #563. 

**Docs**:
I pointed out the difference between gpus=0, gpus=[0] and gpus="0" @jeffling.
Let me know if the column with "Type" is ok or better to be removed.

**Tests**:
I added tests for all inputs listed in the docs, and also tests for everything else that is not currently supported. 

In a follow up PR/Issue, we could discuss the support of the following:

- empty string (no gpus) 
- empty list (no gpus)
- tuples (same behaviour as list)

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

**Comment by awaelchli** [comment]
I think I made a mistake, I should have merged master, not rebase. :(

File: pytorch_lightning/trainer/distrib_parts.py [hunk 1]
```diff
@@ -164,6 +164,37 @@
...
++---------------+-----------+---------------------+---------------------------------+
+| -1            | int       | [0, 1, 2, ...]      | all available GPUs              |
++---------------+-----------+---------------------+---------------------------------+
+| [0]           | list      | [0]                 | GPU 0                           |
++---------------+-----------+---------------------+---------------------------------+
+| [1, 3]        | list      | [1, 3]              | GPUs 1 and 3                    |
++---------------+-----------+---------------------+---------------------------------+
+| "0"           | list      | [0]                 | GPU 0                           |
++---------------+-----------+---------------------+---------------------------------+
+| "3"           | list      | [3]                 | GPU 3                           |
```
Comment (jeffling) [inline pytorch_lightning/trainer/distrib_parts.py]: Should the type be 'str' here?

File: tests/test_gpu_models.py [hunk 2]
```diff
@@ -363,6 +368,33 @@ def test_parse_gpu_ids(mocked_device_count, gpus, expected_gpu_ids):
...
+    pytest.param([]),
+    pytest.param([-1]),
+    pytest.param([None]),
+    pytest.param(['0']),
+    pytest.param((0, 1)),
+]
+
+
+@pytest.mark.gpus_param_tests
+@pytest.mark.parametrize(['gpus'], test_parse_gpu_invalid_inputs_data)
```
Comment (jeffling) [inline tests/test_gpu_models.py]: Nice

File: pytorch_lightning/trainer/distrib_parts.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/trainer/distrib_parts.py]: yes. thanks for checking. fixed

**Comment by awaelchli** [comment]
@Borda I don't understand the error message from travis. Do you know why it fails to install requirements?

**Comment by Borda** [comment]
@awaelchli I am going to check it also discussed at #590

**Pull request merged**